### PR TITLE
[artifactory][artifactory-ha] Explicitly set automountServiceAccountToken to true

### PR DESCRIPTION
**Pull request by andrewsavino1**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:

It is considered best practice (see: CIS EKS Benchmark 4.1.6) to only mount the Service Account token when necessary.  In the case of the artifactory charts the API token is needed, as there is an RBAC role binding to the service account.

**Which issue this PR fixes** 
There is no functional change.  This only adds clarity to the chart.

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by andrewsavino1** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by andrewsavino1** [comment]
recheckcla

**Review by rimusz — APPROVED** [review]
LGTM

File: stable/artifactory-ha/values.yaml [hunk 1]
```diff
@@ -82,6 +82,8 @@ serviceAccount:
   ## If not set and create is true, a name is generated using the fullname template
   name:
   annotations: {}
+  ## explicitly mounts the  API token for the service account to the pods
```
Comment (eldada) [inline stable/artifactory-ha/values.yaml:85]: Can you start with capital `E` please?

**Review by eldada — APPROVED** [review]

File: stable/artifactory-ha/values.yaml (hunk 1 above)
Comment (andrewsavino1) [inline stable/artifactory-ha/values.yaml:85]:
```suggestion
  ## Explicitly mounts the  API token for the service account to the pods
```

File: stable/artifactory/values.yaml [hunk 2]
```diff
@@ -96,6 +96,8 @@ serviceAccount:
   name:
   ## Service Account annotations
   annotations: {}
+  ## explicitly mounts the API credentials for the Service Account
```
Comment (andrewsavino1) [inline stable/artifactory/values.yaml:99]:
```suggestion
  ## Explicitly mounts the API credentials for the Service Account
```

**Comment by andrewsavino1** [comment]
@chukka @rahulsadanandan Would I be able to get a review on this PR?

**Comment by rahulsadanandan** [comment]
@andrewsavino1 Sure, We will test and merge this in the next release of artifactory.

**Comment by dalbertom** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by andrewsavino1** [comment]
@rahulsadanandan Is there any update to this?

**Comment by chukka** [comment]
@andrewsavino1 will be part of 7.18.x release

**Comment by chukka** [comment]
closing this as changes are in #1419

**Pull request closed without merging**